### ISSUE TITLE
Somato Data Filter Question

### ISSUE THREAD

**Issue by tbardouille** [alphacsc/alphacsc#21]
Question about the filter and sample rate on the somato data used for plot somato mu waves. When I look at the info variable, I see that the low-pass is 100Hz and the sample rate is 150Hz. Shouldn't the low-pass be less than 1/2 of the sample rate? 

These settings could be leading to some spurious findings if correct.

Best,
Tim.

**Comment by agramfort** [comment]
Hi Tim,

can you point us to the line that does this? where are you reading this?

Alex

>

**Comment by tomMoral** [comment]
Hi Tim,

Thanks for testing our code and reporting potential mistakes.
The raw data for both `somato` and `sample` have sampling rate of `300Hz` and all preprocessing are performed on these signal to limit potential filtering artifacts and allow us to do filtering up to `150Hz`.
The data is then resampled at `150Hz` after all preprocessing are finished.

I don't think we do any low-pass filtering in our code, could you point us to the section you are referring to? Note that we do use a notch filter [here](https://github.com/alphacsc/alphacsc/blob/master/alphacsc/datasets/somato.py#L55) but this only apply band-cut filters to remove the band `50Hz` and `100Hz` and avoid the powerline artifacts.

**Comment by tbardouille** [comment]
Hi Thomas,

In the code snipper below, you can see that the sample rate is 150 Hz, and that the low-pass filter is 100.1 Hz (in your info variable). This low-pass filter setting is fine for data collected at 300 Hz, because the low-pass filter should be no more than 1/3 of the sample rate.

You should have low-pass filtered the data at 50 Hz before you downsampled to 150 Hz.

Best,
Tim.


In [9]: from alphacsc.datasets.somato import load_data

In [10]: tlim = (-2,4)

In [11]: sfreq=150

In [12]: X, info = load_data(epoch=t_lim, sfreq=sfreq)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-12-b46e8dd860e5> in <module>
----> 1 X, info = load_data(epoch=t_lim, sfreq=sfreq)

NameError: name 't_lim' is not defined

In [13]: X, info = load_data(epoch=tlim, sfreq=sfreq)

In [14]: info
Out[14]:
<Info | 32 non-empty fields
    acq_pars : str | 12853 items
    acq_stim : str | 11999 items
    bads : list | 0 items
    ch_names : list | MEG 0113, MEG 0112, MEG 0122, MEG 0123, MEG 0132, ...
    chs : list | 204 items (GRAD: 204)
    comps : list | 0 items
    cov : Covariance | 13 items
    custom_ref_applied : bool | False
    description : str | 45 items
    dev_head_t : Transform | 3 items
    dig : list | 47 items (3 Cardinal, 4 HPI, 40 Extra)
    event_id : int | 1
    events : ndarray | 111 items
    experimenter : str | 4 items
    file_bem : str | 79 items
    file_id : dict | 4 items
    file_trans : str | 68 items
    highpass : float | 2.0 Hz
    hpi_meas : list | 1 items
    hpi_results : list | 1 items
    lowpass : float | 100.10250091552734 Hz
    meas_date : tuple | 2007-07-05 11:17:11 GMT
    meas_id : dict | 4 items
    nchan : int | 204
    proc_history : list | 1 items
    proj_id : ndarray | 1 items
    proj_name : str | 9 items
    projs : list | 0 items
    sfreq : float | 150.0 Hz
    subject : str | 6 items
    subject_info : dict | 1 items
    subjects_dir : str | 44 items
    ctf_head_t : NoneType
    dev_ctf_t : NoneType
    gantry_angle : NoneType
    hpi_subsystem : NoneType
    kit_system_id : NoneType
    line_freq : NoneType
    xplotter_layout : NoneType
>

**Comment by TomDLT** [comment]
The `info` dictionary contains this lowpass entry even in the raw data, before resampling at 150 Hz.
This is unrelated to the resampling at 150 Hz.

```py
import mne
from os.path import join

data_path = mne.datasets.somato.data_path()
data_dir = join(data_path, 'MEG', 'somato')
file_name = join(data_dir, 'sef_raw_sss.fif')
raw = mne.io.read_raw_fif(file_name, preload=True, verbose=0)

print(raw.info['lowpass'])
```

However, the resampling also performs a low-pass filtering step at `sfreq / 2.0` to avoid aliasing (see [FAQ](http://martinos.org/mne/stable/faq.html?highlight=faq#resampling-and-decimating-data)). The entry `info['lowpass']` [is updated](https://github.com/mne-tools/mne-python/blob/d36440176cf3f3532f64e6f046c4a6a3eca028de/mne/io/base.py#L1407) in `Raw.resample` but [is not updated](https://github.com/mne-tools/mne-python/blob/d36440176cf3f3532f64e6f046c4a6a3eca028de/mne/filter.py#L1952) in `Epochs.resample`. I opened an issue upstream (https://github.com/mne-tools/mne-python/issues/6468).

**Comment by TomDLT** [comment]
This bug was fixed upstream (mne-tools/mne-python#6468).
Thanks for the report.

**Comment by tbardouille** [comment]
Thanks.
